model/src/main/java/com/redhat/ipaas/model/integration/Integration.java

```diff
@@ -65,6 +65,8 @@ default Kind getKind() {
 
     Optional<Status> getCurrentStatus();
 
+    Optional<List<String>> getStepsDone();
```

Comment:
Do you mean a separately persisted object or a nested object? I was meaning nested objects, `integration.spec` and `integration.status` that define desired and current state respectively so you'd have `integration.status.currentStatus` and `integration.status.stepsDone` (ignore naming) to describe current state. We will look at this next week, post-summit so I wouldn't spend too much time thinking about this particular PR, if it gets past the immediate issue.